pmd-core/src/main/java/net/sourceforge/pmd/lang/AntlrTokenManager.java

```diff
@@ -0,0 +1,79 @@
+/**
+ * BSD-style license; for more info see http://pmd.sourceforge.net/license.html
+ */
+
+package net.sourceforge.pmd.lang;
+
+import org.antlr.v4.runtime.BaseErrorListener;
+import org.antlr.v4.runtime.Lexer;
+import org.antlr.v4.runtime.RecognitionException;
+import org.antlr.v4.runtime.Recognizer;
+
+/**
+ * Generic token manager implementation for all Antlr lexers.
+ */
+public class AntlrTokenManager implements TokenManager {
+    private final Lexer lexer;
+    private String fileName;
+
+    /**
+     * Constructor
+     *
+     * @param lexer The lexer
+     * @param fileName The file name
+     */
+    public AntlrTokenManager(final Lexer lexer, final String fileName) {
+        this.lexer = lexer;
+        this.fileName = fileName;
+    }
+
+    @Override
+    public Object getNextToken() {
+        return lexer.nextToken();
+    }
+
+    @Override
+    public void setFileName(String fileName) {
+        this.fileName = fileName;
+    }
+
+    public String getFileName() {
+        return fileName;
+    }
+
+    public void resetListeners() {
```

Comment:
is this really needed as a public method? shouldn't it be part of the constructor?

Reply:
Agreed, we will change this